cmd/telegraf/telegraf.go

```diff
@@ -117,44 +124,46 @@ func reloadLoop(
 func runAgent(ctx context.Context,
 	inputFilters []string,
 	outputFilters []string,
-) error {
+) (*agent.Agent, error) {
+	var ag *agent.Agent
```

Comment:
This line will shadow the global variable. In general I don't see the point of returning the agent only to afterwards set a global variable to this return value. You can cook down the whole patch quite a bit if you set the global variable in here directly.